lib/react.dart

```diff
@@ -1766,6 +1791,18 @@ ComponentRegistrar2 registerComponent2 = (
   throw new Exception('setClientConfiguration must be called before registerComponent.');
 };
 
+/// Registers [componentFactory] on client.
+///
+/// Example:
+/// ```
+///   var myFunctionComponent = registerFunctionComponent((Map props) {
```

Comment:
Should the example show declaring it as a separate function or passing in `displayName` so that there's a displayName built in?